### PR TITLE
finished with lab

### PR DESCRIPTION
**Pull request by devonhackley**

### CODE REVIEW

**Comment by devonhackley** [comment]
### Linked list
- [ x] head
- [x] insert
- [x] includes
- [x] print
- [x] read me

File: code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package challenges.linkedlist;
+
+import java.util.ArrayList;
+
+public class LinkedList {
+   public  Node head;
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java]: Not significant but there are two spaces between `public` and `Node`

File: code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   public int length;
+
+    public void LinkedList(){
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java]: The `void` is not necessary and is technically not true. This is the constructor which is returning something, an instance of the LinkedList class

File: code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    public void insert(int data){
+        try
+        {
+            Node node = new Node(data);
+            node.next = null;
+            if(this.head == null){
+                this.head = node;
+                this.length++;
+            } else {
+                Node current = this.head;
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java]: If line 23 was `node.next = this.head` then you could get rid of this line completely

File: code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            }
+        }
+        catch (IllegalArgumentException ex)
+        {
+            System.out.println("You either didn't pass anything in or you didn't pass in an Integer");
+        }
+        return isIncluded;
+    }
+
+    public ArrayList print(){
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java]: It's good practice to specify the data type an ArrayList will have, in this case:
`ArrayList<Integer>`

File: code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        boolean isIncluded = false;
+        try
+        {
+            while(current != null){
+                if(current.data == data){
+                    isIncluded = true;
+                }
+                current = current.next;
+            }
+        }
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java]: Per standard Java syntax the catch should be on the same line as the closing `}` of the `try`

`} catch (IllegalArgumentException ex) {`

This isn't required for the code to work though

File: code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+public class LinkedList {
+   public  Node head;
+   public int length;
+
+    public void LinkedList(){
+            this.head = null;
+            this.length = 0;
+    }
+    public void insert(int data){
+        try
```
Comment (evanslaton) [inline code-challenges/401challenges/src/main/java/challenges/linkedlist/LinkedList.java]: The `{` should be on the same line as the `try`

`try {`

**Pull request merged**